src/librustc_middle/ty/print/pretty.rs

```diff
@@ -1317,6 +1346,13 @@ impl<F: fmt::Write> Printer<'tcx> for FmtPrinter<'_, 'tcx, F> {
         define_scoped_cx!(self);
 
         if substs.is_empty() {
+            if !SHOULD_PREFIX_WITH_CRATE.with(|flag| flag.get()) {
+                match self.try_print_unique_def(def_id)? {
+                    (cx, true) => return Ok(cx),
+                    (cx, false) => self = cx,
+                }
+            }
+
             match self.try_print_visible_def_path(def_id)? {
```

Comment:
I wonder if it might be useful to move the logic into `try_print_visible_def_path`.

Reply:
I thought so to, but figured that the name of `try_print_visible_def_path` would need to change to convey its dual purpose, and the coupled name name `try_print_visible_def_path_recur` too, so I'm not sure about it. Given that `try_print_visible_def_path` has a single call site, the benefit feels diminished to me.

Reply:
But "visible def path" is what you're doing, isn't it? Just not absolute. Maybe I'm overthinking things.

Reply:
Seems ok as is, marking as resolved.